cactus-maven-plugin/src/main/java/com/telenav/cactus/maven/GitMergePullRequestMojo.java

```diff
@@ -0,0 +1,320 @@
+
+    /**
+     * If true, pass <code>--auto</code> to `gh pr merge`.
+     */
+    @Parameter(property = "cactus.pr.auto", defaultValue = "false")
+    private boolean auto;
+
+    /**
+     * If true, pass <code>--delete-branch</code> to `gh pr merge`.
+     */
+    @Parameter(property = "cactus.pr.delete-branch", defaultValue = "false")
+    private boolean deleteBranch;
+
+    /**
+     * If true, pass <code>--merge</code> to `gh pr merge`. Mutually exclusive
+     * with the rebase option.
+     */
+    @Parameter(property = "cactus.pr.merge", defaultValue = "true")
+    private boolean merge;
+
+    /**
+     * If true, pass <code>--squash</code> to `gh pr merge`,
+     */
+    @Parameter(property = "cactus.pr.squash", defaultValue = "false")
+    private boolean squash;
+
+    /**
+     * If true, pass <code>--rebase</code> to `gh pr merge`. Mutually exclusive
+     * with the merge option.
+     */
+    @Parameter(property = "cactus.pr.rebase", defaultValue = "false")
+    private boolean rebase;
+
+    /**
+     * If true, pass <code>--auto</code> to `gh pr merge`.
+     */
+    @Parameter(property = "cactus.pr.admin", defaultValue = "false")
+    private boolean admin;
+
+    /**
+     * The base branch we intend the PR to be merged to - this is not passed to
+     * the client, but to weed out checkouts which are on it and therefore not
+     * likely to be part of the stuff to merge. The default is `develop`.
+     */
+    @Parameter(property = "cactus.base-branch", defaultValue = "develop")
+    private String baseBranch = "develop";
+
+    public GitMergePullRequestMojo()
+    {
+        super(RunPolicies.LAST);
+    }
+
+    @Override
+    protected void onValidateGithubParameters(BuildLog log, MavenProject project)
+            throws Exception
+    {
+        validateBranchName(targetBranch, true);
+        validateBranchName(baseBranch, false);
+        Set<MergePullRequestOptions> opts = options();
+        Set<MergePullRequestOptions> cannotBeCombined
+                = EnumSet.of(MERGE, SQUASH, REBASE);
+        cannotBeCombined.retainAll(opts);
+        if (cannotBeCombined.size() > 1)
+        {
+            fail("Only one of MERGE, REBASE and SQUASH may be set to true, but "
+                    + "have the options " + opts);
+        }
+    }
+
+    @Override
+    protected void execute(BuildLog log, MavenProject project,
+            GitCheckout myCheckout,
+            ProjectTree tree, List<GitCheckout> checkouts) throws Exception
+    {
+        if (checkouts.isEmpty())
+        {
+            log.info("No checkouts matched.");
+            return;
+        }
+        String branch = targetBranch(myCheckout, tree);
+        if (branch.equals(baseBranch))
+        {
+            fail("The target branch and the base branch are both '" + branch
+                    + "'.  There will not be any PRs from a branch to itself.");
+        }
+
+        Map<GitCheckout, MinimalPRItem> prForCheckout = findInitialPR(log,
```

Comment:
throughout, the `var` keyword would make for easier reading

Reply:
As long as you know that a `Map` is what `findInitialPR`.

Seriously, watching you struggle with cases where an API change broke the invisible type encoded by `var`, I am increasingly convinced that it makes absolutely nothing better.